Library/Homebrew/utils/gzip.rb

```diff
@@ -28,11 +28,11 @@ def compress_with_options(path, mtime: ENV["SOURCE_DATE_EPOCH"].to_i, orig_name:
       # Zlib::GzipWriter does not properly handle the case of setting mtime = 0:
       # https://bugs.ruby-lang.org/issues/16285
       #
-      # This was fixed in https://github.com/ruby/zlib/pull/10. Set mtime to 0 instead
-      # of raising exception once we are using zlib gem version 1.1.0 or newer.
+      # This was fixed in https://github.com/ruby/zlib/pull/10. Remove workaround
+      # once we are using zlib gem version 1.1.0 or newer.
```

Comment:
How many bottles does this affect in practice? If it's just one or two that's getting (semi-)frequent bottle updates anyway then it's probably ok.

I hope we're not stuck with Ruby 2.6 for _that_ long (Ruby 2.7 has the bugfix) but I guess we'll see come June.

Reply:
@carlocab Does this change the checksum of any existing bottles?

> I hope we're not stuck with Ruby 2.6 for _that_ long (Ruby 2.7 has the bugfix) but I guess we'll see come June.

I think we could bump the requirement for CI/dev-cmd before that but probably want to assume end-users may be running it for a while longer.

Reply:
> So far that I know of? Only one at [Homebrew/homebrew-core#123724](https://github.com/Homebrew/homebrew-core/pull/123724). But throwing an error here is relatively new, so there could be more. Presumably @alebcay spotted some others which is what prompted throwing an error here.

I noticed this because my initial sanity testing with `GzipWriter` (in `brew irb`) was with `mtime` of 0. That's how I discovered the corner case.

I actually didn't know of any actual need we had for setting `mtime` of 0, this is the first I'm hearing of it.

Reply:
> @carlocab Does this change the checksum of any existing bottles?

If it does, it would be because their checksums were not reproducible. @alebcay also [points this out](https://github.com/Homebrew/brew/pull/14723#issuecomment-1437486178). Making this change will make their checksums reproducible (as long as we never remove the bit that does `mtime = 1 if mtime.zero?`).